src/bokeh/models/glyphs.py

```diff
@@ -1071,6 +1071,65 @@ def __init__(self, *args, **kwargs) -> None:
     The {prop} values for the multipolygons.
     """)
 
+class Ngon(RadialGlyph):
+    ''' Render regular n-sided polygons.
+
+    '''
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    __example__ = "examples/reference/models/Ngon.py"
+
+    _args = ('x', 'y', 'radius')
+
+    x = NumberSpec(default=field("x"), help="""
+    The x-coordinates of the center of the n-gons.
+    """)
```

Comment:
ngon has some additional specific requirements to note, i.e. that the radius measures to a vertex, rather than, say, to the  middle of an edge, and that a vertex is always located directly above the point, at angle=0. 

Beyond that, while I definitely understand the motivation to "factor" things as much as possible for our sake, for users' it is better to actually be as specific as possible everywhere. So I am 👎 on making any changes to the way things are in the PR.